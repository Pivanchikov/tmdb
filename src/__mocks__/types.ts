import { Movie } from '../state/types'

export type Genres = {
  id: number
  name: string
}

export type GenresSelectOption = {
  value: number | string
  label: string
}

export type TransformMovies = {
  [key: string]: Movie
}
