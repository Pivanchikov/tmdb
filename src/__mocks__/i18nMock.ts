export default jest.mock('react-i18next', () => {
  return {
    useTranslation: () => {
      return {
        t: jest.fn(value => value),
        i18n: {
          language: 'en',
          changeLanguage: jest.fn(),
        },
      }
    },
  }
})
