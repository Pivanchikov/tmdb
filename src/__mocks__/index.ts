export * from './mockGenres'
export * from './mockMovies'
export * from './storeMock'
