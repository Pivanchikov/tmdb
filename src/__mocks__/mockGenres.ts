import { Genres, GenresSelectOption } from './types'

export const enGenresMock: Genres[] = [
  { id: 28, name: 'Action' },
  { id: 12, name: 'Adventure' },
  { id: 16, name: 'Animation' },
  { id: 35, name: 'Comedy' },
  { id: 80, name: 'Crime' },
  { id: 99, name: 'Documentary' },
  { id: 18, name: 'Drama' },
  { id: 10751, name: 'Family' },
  { id: 14, name: 'Fantasy' },
  { id: 36, name: 'History' },
  { id: 27, name: 'Horror' },
  { id: 10402, name: 'Music' },
  { id: 9648, name: 'Mystery' },
  { id: 10749, name: 'Romance' },
  { id: 878, name: 'Science Fiction' },
  { id: 10770, name: 'TV Movie' },
  { id: 53, name: 'Thriller' },
  { id: 10752, name: 'War' },
  { id: 37, name: 'Western' },
]

export const enGenresSelectMock: GenresSelectOption[] = [
  { value: 28, label: 'Action' },
  { value: 12, label: 'Adventure' },
  { value: 16, label: 'Animation' },
  { value: 35, label: 'Comedy' },
  { value: 80, label: 'Crime' },
  { value: 99, label: 'Documentary' },
  { value: 18, label: 'Drama' },
  { value: 10751, label: 'Family' },
  { value: 14, label: 'Fantasy' },
  { value: 36, label: 'History' },
  { value: 27, label: 'Horror' },
  { value: 10402, label: 'Music' },
  { value: 9648, label: 'Mystery' },
  { value: 10749, label: 'Romance' },
  { value: 878, label: 'Science Fiction' },
  { value: 10770, label: 'TV Movie' },
  { value: 53, label: 'Thriller' },
  { value: 10752, label: 'War' },
  { value: 37, label: 'Western' },
]
export const ruGenresMock: Genres[] = [
  { id: 28, name: 'боевик' },
  { id: 12, name: 'приключения' },
  { id: 16, name: 'мультфильм' },
  { id: 35, name: 'комедия' },
  { id: 80, name: 'криминал' },
  { id: 99, name: 'документальный' },
  { id: 18, name: 'драма' },
  { id: 10751, name: 'семейный' },
  { id: 14, name: 'фэнтези' },
  { id: 36, name: 'история' },
  { id: 27, name: 'ужасы' },
  { id: 10402, name: 'музыка' },
  { id: 9648, name: 'детектив' },
  { id: 10749, name: 'мелодрама' },
  { id: 878, name: 'фантастика' },
  { id: 10770, name: 'телевизионный фильм' },
  { id: 53, name: 'триллер' },
  { id: 10752, name: 'военный' },
  { id: 37, name: 'вестерн' },
]

export const emptyGenre: GenresSelectOption = { value: '', label: 'Any genre' }

export const ruGenresSelectMock: GenresSelectOption[] = [
  { value: 28, label: 'Боевик' },
  { value: 12, label: 'Приключения' },
  { value: 16, label: 'Мультфильм' },
  { value: 35, label: 'Комедия' },
  { value: 80, label: 'Криминал' },
  { value: 99, label: 'Документальный' },
  { value: 18, label: 'Драма' },
  { value: 10751, label: 'Семейный' },
  { value: 14, label: 'Фэнтези' },
  { value: 36, label: 'История' },
  { value: 27, label: 'Ужасы' },
  { value: 10402, label: 'Музыка' },
  { value: 9648, label: 'Детектив' },
  { value: 10749, label: 'Мелодрама' },
  { value: 878, label: 'Фантастика' },
  { value: 10770, label: 'Телевизионный фильм' },
  { value: 53, label: 'Триллер' },
  { value: 10752, label: 'Военный' },
  { value: 37, label: 'Вестерн' },
]
