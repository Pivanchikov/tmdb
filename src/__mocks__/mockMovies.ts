/* eslint-disable no-magic-numbers */
import { TransformMovies } from './types'
import { Movie } from '../state/types'

export const movieWithAllParam: Movie = {
  adult: false,
  backdrop_path: '/las0P4Dua54XrZ73VQmGUaH1z0U.jpg',
  genre_ids: [16, 28, 18, 878],
  id: 283566,
  original_language: 'ja',
  original_title: 'シン・エヴァンゲリオン劇場版:||',
  overview:
    'In the aftermath of the Fourth Impact, stranded without their Evangelions, Shinji, Asuka, and Rei search for refuge in the desolate red remains of Tokyo-3. But the danger to the world is far from over. A new impact is looming on the horizon—one that will prove to be the true end of Evangelion.',
  popularity: 1125.051,
  poster_path: '/jDwZavHo99JtGsCyRzp4epeeBHx.jpg',
  release_date: '2021-03-08',
  title: 'Evangelion: 3.0+1.0 Thrice Upon a Time',
  video: false,
  vote_average: 8.8,
  vote_count: 194,
}

export const movieWithOutAllParam: Movie = {
  adult: false,
  backdrop_path: '/las0P4Dua54XrZ73VQmGUaH1z0U.jpg',
  genre_ids: [16, 28, 18, 878],
  id: 283566,
  original_language: 'ja',
  original_title: 'シン・エヴァンゲリオン劇場版:||',
  overview: null,
  popularity: 1125.051,
  poster_path: null,
  release_date: '2021-03-08',
  title: 'Evangelion: 3.0+1.0 Thrice Upon a Time',
  video: false,
  vote_average: 8,
  vote_count: 194,
}

export const movies: Movie[] = [
  {
    adult: false,
    backdrop_path: '/las0P4Dua54XrZ73VQmGUaH1z0U.jpg',
    genre_ids: [16, 28, 18, 878],
    id: 283566,
    original_language: 'ja',
    original_title: 'シン・エヴァンゲリオン劇場版:||',
    overview:
      'In the aftermath of the Fourth Impact, stranded without their Evangelions, Shinji, Asuka, and Rei search for refuge in the desolate red remains of Tokyo-3. But the danger to the world is far from over. A new impact is looming on the horizon—one that will prove to be the true end of Evangelion.',
    popularity: 1125.051,
    poster_path: '/jDwZavHo99JtGsCyRzp4epeeBHx.jpg',
    release_date: '2021-03-08',
    title: 'Evangelion: 3.0+1.0 Thrice Upon a Time',
    video: false,
    vote_average: 8.8,
    vote_count: 194,
  },
  {
    adult: false,
    backdrop_path: '/6MKr3KgOLmzOP6MSuZERO41Lpkt.jpg',
    genre_ids: [35, 80],
    id: 337404,
    original_language: 'en',
    original_title: 'Cruella',
    overview:
      'In 1970s London amidst the punk rock revolution, a young grifter named Estella is determined to make a name for herself with her designs. She befriends a pair of young thieves who appreciate her appetite for mischief, and together they are able to build a life for themselves on the London streets. One day, Estella’s flair for fashion catches the eye of the Baroness von Hellman, a fashion legend who is devastatingly chic and terrifyingly haute. But their relationship sets in motion a course of events and revelations that will cause Estella to embrace her wicked side and become the raucous, fashionable and revenge-bent Cruella.',
    popularity: 978.832,
    poster_path: '/wToO8opxkGwKgSfJ1JK8tGvkG6U.jpg',
    release_date: '2021-05-26',
    title: 'Cruella',
    video: false,
    vote_average: 8.4,
    vote_count: 4777,
  },
  {
    adult: false,
    backdrop_path: '/8s4h9friP6Ci3adRGahHARVd76E.jpg',
    genre_ids: [16, 35, 10751, 878],
    id: 379686,
    original_language: 'en',
    original_title: 'Space Jam: A New Legacy',
    overview:
      "When LeBron and his young son Dom are trapped in a digital space by a rogue A.I., LeBron must get them home safe by leading Bugs, Lola Bunny and the whole gang of notoriously undisciplined Looney Tunes to victory over the A.I.'s digitized champions on the court. It's Tunes versus Goons in the highest-stakes challenge of his life.",
    popularity: 2105.173,
    poster_path: '/5bFK5d3mVTAvBCXi5NPWH0tYjKl.jpg',
    release_date: '2021-07-08',
    title: 'Space Jam: A New Legacy',
    video: false,
    vote_average: 7.5,
    vote_count: 1774,
  },
  {
    adult: false,
    backdrop_path: '/xXHZeb1yhJvnSHPzZDqee0zfMb6.jpg',
    genre_ids: [28, 80, 53],
    id: 385128,
    original_language: 'en',
    original_title: 'F9',
    overview:
      "Dominic Toretto and his crew battle the most skilled assassin and high-performance driver they've ever encountered: his forsaken brother.",
    popularity: 2119.117,
    poster_path: '/bOFaAXmWWXC3Rbv4u4uM9ZSzRXP.jpg',
    release_date: '2021-05-19',
    title: 'F9',
    video: false,
    vote_average: 7.7,
    vote_count: 2804,
  },
  {
    adult: false,
    backdrop_path: '/jlGmlFOcfo8n5tURmhC7YVd4Iyy.jpg',
    genre_ids: [28, 12, 14],
    id: 436969,
    original_language: 'en',
    original_title: 'The Suicide Squad',
    overview:
      'Supervillains Harley Quinn, Bloodsport, Peacemaker and a collection of nutty cons at Belle Reve prison join the super-secret, super-shady Task Force X as they are dropped off at the remote, enemy-infused island of Corto Maltese.',
    popularity: 6294.822,
    poster_path: '/iCi4c4FvVdbaU1t8poH1gvzT6xM.jpg',
    release_date: '2021-07-28',
    title: 'The Suicide Squad',
    video: false,
    vote_average: 8.1,
    vote_count: 2435,
  },
]

export const transformMovies: TransformMovies = {
  '283566': {
    adult: false,
    backdrop_path: '/las0P4Dua54XrZ73VQmGUaH1z0U.jpg',
    genre_ids: [16, 28, 18, 878],
    id: 283566,
    original_language: 'ja',
    original_title: 'シン・エヴァンゲリオン劇場版:||',
    overview:
      'In the aftermath of the Fourth Impact, stranded without their Evangelions, Shinji, Asuka, and Rei search for refuge in the desolate red remains of Tokyo-3. But the danger to the world is far from over. A new impact is looming on the horizon—one that will prove to be the true end of Evangelion.',
    popularity: 1125.051,
    poster_path: '/jDwZavHo99JtGsCyRzp4epeeBHx.jpg',
    release_date: '2021-03-08',
    title: 'Evangelion: 3.0+1.0 Thrice Upon a Time',
    video: false,
    vote_average: 8.8,
    vote_count: 194,
  },
  '337404': {
    adult: false,
    backdrop_path: '/6MKr3KgOLmzOP6MSuZERO41Lpkt.jpg',
    genre_ids: [35, 80],
    id: 337404,
    original_language: 'en',
    original_title: 'Cruella',
    overview:
      'In 1970s London amidst the punk rock revolution, a young grifter named Estella is determined to make a name for herself with her designs. She befriends a pair of young thieves who appreciate her appetite for mischief, and together they are able to build a life for themselves on the London streets. One day, Estella’s flair for fashion catches the eye of the Baroness von Hellman, a fashion legend who is devastatingly chic and terrifyingly haute. But their relationship sets in motion a course of events and revelations that will cause Estella to embrace her wicked side and become the raucous, fashionable and revenge-bent Cruella.',
    popularity: 978.832,
    poster_path: '/wToO8opxkGwKgSfJ1JK8tGvkG6U.jpg',
    release_date: '2021-05-26',
    title: 'Cruella',
    video: false,
    vote_average: 8.4,
    vote_count: 4777,
  },
  '379686': {
    adult: false,
    backdrop_path: '/8s4h9friP6Ci3adRGahHARVd76E.jpg',
    genre_ids: [16, 35, 10751, 878],
    id: 379686,
    original_language: 'en',
    original_title: 'Space Jam: A New Legacy',
    overview:
      "When LeBron and his young son Dom are trapped in a digital space by a rogue A.I., LeBron must get them home safe by leading Bugs, Lola Bunny and the whole gang of notoriously undisciplined Looney Tunes to victory over the A.I.'s digitized champions on the court. It's Tunes versus Goons in the highest-stakes challenge of his life.",
    popularity: 2105.173,
    poster_path: '/5bFK5d3mVTAvBCXi5NPWH0tYjKl.jpg',
    release_date: '2021-07-08',
    title: 'Space Jam: A New Legacy',
    video: false,
    vote_average: 7.5,
    vote_count: 1774,
  },
  '385128': {
    adult: false,
    backdrop_path: '/xXHZeb1yhJvnSHPzZDqee0zfMb6.jpg',
    genre_ids: [28, 80, 53],
    id: 385128,
    original_language: 'en',
    original_title: 'F9',
    overview:
      "Dominic Toretto and his crew battle the most skilled assassin and high-performance driver they've ever encountered: his forsaken brother.",
    popularity: 2119.117,
    poster_path: '/bOFaAXmWWXC3Rbv4u4uM9ZSzRXP.jpg',
    release_date: '2021-05-19',
    title: 'F9',
    video: false,
    vote_average: 7.7,
    vote_count: 2804,
  },
  '436969': {
    adult: false,
    backdrop_path: '/jlGmlFOcfo8n5tURmhC7YVd4Iyy.jpg',
    genre_ids: [28, 12, 14],
    id: 436969,
    original_language: 'en',
    original_title: 'The Suicide Squad',
    overview:
      'Supervillains Harley Quinn, Bloodsport, Peacemaker and a collection of nutty cons at Belle Reve prison join the super-secret, super-shady Task Force X as they are dropped off at the remote, enemy-infused island of Corto Maltese.',
    popularity: 6294.822,
    poster_path: '/iCi4c4FvVdbaU1t8poH1gvzT6xM.jpg',
    release_date: '2021-07-28',
    title: 'The Suicide Squad',
    video: false,
    vote_average: 8.1,
    vote_count: 2435,
  },
}

export const mockMovies = [[movies, transformMovies]]
