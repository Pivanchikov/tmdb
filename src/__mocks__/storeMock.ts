import { MoviesState } from '../state/types'

export const initialState: MoviesState = {
  items: [],
  byId: {},
  genres: [],
  selectedMovie: null,
  isLoading: false,
  isFoundMovie: false,
  popularMovies: [],
  trendingMovies: [],
  nowPlayingMovies: [],
  upcomingMovies: [],
  topRatedMovies: [],
}
