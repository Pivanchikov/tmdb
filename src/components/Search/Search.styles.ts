import { desktopWidth } from '../constants'
import { primaryMain, primaryLight, primaryLightHover, secondaryMain } from '../../theme/theme'

export const styles = theme => ({
  paper: {
    minHeight: 64,
    backgroundColor: theme.palette.primary.main,
    display: 'flex',
    justifyContent: 'space-around',
    alignItems: 'baseline',
    // eslint-disable-next-line @typescript-eslint/prefer-as-const
    flexDirection: 'row' as 'row',
    padding: '10px 15px',
  },
  searchContainer: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    width: '25%',
  },
  iconButton: {
    color: theme.palette.secondary.main,
  },
})

export const select = {
  container: base => ({
    ...base,
    minWidth: 220,
    width: innerWidth < desktopWidth ? '80%' : '90%',
  }),

  control: (base, state) => ({
    ...base,
    borderColor: state.isFocused ? primaryLightHover : primaryLight,
    boxShadow: null,
    '&:hover': {
      borderColor: primaryLightHover,
    },
  }),

  menu: base => ({
    ...base,
    position: innerWidth < desktopWidth ? 'relative' : 'absolute',
  }),
  option: base => ({
    ...base,
    color: secondaryMain,
  }),
  dropdownIndicator: base => ({
    ...base,
    color: primaryMain,
    '&:hover': { color: primaryMain },
  }),
  placeholder: base => ({
    ...base,
    color: secondaryMain,
  }),
  indicatorSeparator: base => ({
    ...base,
    backgroundColor: primaryMain,
  }),
}

export const selectTheme = theme => ({
  ...theme,
  colors: {
    ...theme.colors,
    primary25: primaryLightHover,
    primary: primaryLight,
    neutral0: primaryMain,
    neutral80: secondaryMain,
  },
})
