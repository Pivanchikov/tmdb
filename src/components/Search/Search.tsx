import { useState, useEffect, useMemo } from 'react'
import { IconButton, Drawer, makeStyles } from '@material-ui/core'
import SearchIcon from '@material-ui/icons/Search'
import CloseIcon from '@material-ui/icons/Close'
import Select from 'react-select'
import { useTranslation } from 'react-i18next'
import { useSelector, useDispatch } from 'react-redux'
import { Redirect } from 'react-router-dom'
import { debounce } from 'lodash'

import { desktopWidth, TIMEOUT_MS, QUERY_LENGTH_THRESHOLD } from '../constants'
import { APP_ROUTES } from '../../routes'
import { styles, select, selectTheme } from './Search.styles'
import { getMoviesItems, fetchMovies } from '../../state'

const useStyles = makeStyles(styles)

export const Search = () => {
  const dispatch = useDispatch()
  const { t } = useTranslation()
  const classes = useStyles()
  const [redirectSearch, setRedirectSearch] = useState(false)
  const [selectedMovieId, setSelectedMovieId] = useState('')
  const [searchString, setSearchString] = useState('')
  const [opened, setOpened] = useState(false)

  const movies = useSelector(getMoviesItems)

  const options = useMemo(() => movies.map(item => ({ value: item.id, label: item.title })), [movies])

  const handleToggle = () => setOpened(state => !state)
  const openSearchPage = () => setRedirectSearch(state => !state)

  const handleChange = e => {
    handleToggle()
    setSelectedMovieId(`${e.value}-${e.label}`)
  }

  const handleInputChange = (value: string) => setSearchString(value)

  const doSearch = useMemo(
    () =>
      debounce((value: string) => {
        dispatch(fetchMovies(value))
      }, TIMEOUT_MS),
    [dispatch]
  )

  useEffect(() => {
    if (searchString.length < QUERY_LENGTH_THRESHOLD) return
    doSearch(searchString)
  }, [searchString, doSearch])

  return innerWidth < desktopWidth ? (
    <>
      {selectedMovieId && <Redirect to={`/movie/${selectedMovieId}`} />}
      {redirectSearch && <Redirect to={APP_ROUTES.DISCOVER_PAGE} />}
      <IconButton onClick={handleToggle} data-testid="buttonSearch" className={classes.iconButton}>
        <SearchIcon fontSize="medium" />
      </IconButton>
      <Drawer anchor="top" open={opened} onClose={handleToggle} classes={{ paper: classes.paper }}>
        <IconButton onClick={openSearchPage} data-testid="buttonSearchPage" className={classes.iconButton}>
          <SearchIcon fontSize="medium" data-testid="searchIcon" />
        </IconButton>
        <Select
          options={options}
          theme={selectTheme}
          styles={select}
          placeholder={t('search')}
          onChange={handleChange}
          onInputChange={handleInputChange}
          data-testid="SearchMobile"
        />
        <IconButton onClick={handleToggle} data-testid="closeDrawer" className={classes.iconButton}>
          <CloseIcon fontSize="medium" />
        </IconButton>
      </Drawer>
    </>
  ) : (
    <>
      {selectedMovieId && <Redirect to={`/movie/${selectedMovieId}`} />}
      {redirectSearch && <Redirect to={APP_ROUTES.DISCOVER_PAGE} />}
      <div className={classes.searchContainer}>
        <Select
          options={options}
          theme={selectTheme}
          styles={select}
          placeholder={t('search')}
          onChange={handleChange}
          onInputChange={handleInputChange}
          name="Search"
        />
        <IconButton onClick={openSearchPage} data-testid="buttonSearch" className={classes.iconButton}>
          <SearchIcon fontSize="medium" data-testid="searchIcon" />
        </IconButton>
      </div>
    </>
  )
}
