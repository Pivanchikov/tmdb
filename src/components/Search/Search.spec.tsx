/* eslint-disable react/prop-types */
/* eslint-disable no-global-assign */
/* eslint-disable react/display-name */

import { render as testingLibraryRender, fireEvent } from '@testing-library/react'
import { Provider } from 'react-redux'
import React from 'react'
import configureStore from 'redux-mock-store'
import { createMemoryHistory } from 'history'
import { Router } from 'react-router-dom'

import { Search } from './Search'
import { movies, transformMovies } from '../../__mocks__/mockMovies'

const mockStore = configureStore([])

jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: jest.fn(value => value),
    i18n: {
      language: 'en',
      changeLanguage: jest.fn(),
    },
  }),
}))

jest.mock('react-select', () => ({ options, value, onChange }) => {
  function handleChange(event) {
    const option = options.find(option => {
      return option.value == event.currentTarget.value
    })

    onChange(option)
  }

  return (
    <select id="uc" data-testid="select" value={value} onChange={event => handleChange(event)}>
      {options?.map(({ label, value }) => (
        <option key={value} value={value}>
          {label}
        </option>
      ))}
    </select>
  )
})

describe('SearchDecstop', () => {
  let store

  beforeEach(() => {
    jest.resetModules()
    store = mockStore({
      movies: { items: movies, byId: transformMovies },
    })
  })

  const render = () =>
    testingLibraryRender(
      <Router history={createMemoryHistory()}>
        <Provider store={store}>
          <Search />
        </Provider>
      </Router>
    )

  it('renders correctly initial document (desktop version)', () => {
    window = Object.assign(window, { innerWidth: 1500 })
    const { getByTestId } = render()
    expect(getByTestId('select')).toBeInTheDocument
  })

  it('renders correctly initial document (mobile version)', () => {
    window = Object.assign(window, { innerWidth: 500 })
    const { getByTestId } = render()

    expect(getByTestId('buttonSearch')).toBeInTheDocument

    fireEvent.click(getByTestId('buttonSearch'))

    expect(getByTestId('searchIcon')).toBeInTheDocument
    expect(getByTestId('select')).toBeInTheDocument

    expect(getByTestId('closeDrawer')).toBeInTheDocument

    fireEvent.click(getByTestId('closeDrawer'))

    expect(getByTestId('searchIcon')).not.toBeInTheDocument
    expect(getByTestId('select')).toBeInTheDocument
    expect(getByTestId('closeDrawer')).not.toBeInTheDocument
  })

  it('renders correctly redirect to search page from mobile page', () => {
    const { getByTestId } = render()

    expect(getByTestId('buttonSearch')).toBeInTheDocument

    fireEvent.click(getByTestId('buttonSearch'))

    fireEvent.click(getByTestId('searchIcon'))
  })

  it('renders correctly redirect to search page from desktop page', () => {
    window = Object.assign(window, { innerWidth: 1500 })
    const { getByTestId } = render()

    expect(getByTestId('buttonSearch')).toBeInTheDocument

    fireEvent.click(getByTestId('buttonSearch'))
  })

  it('checking redirect to  movie page', () => {
    window = Object.assign(window, { innerWidth: 500 })
    React.useState = jest.fn().mockReturnValue([true, {}])
    render()

    window = Object.assign(window, { innerWidth: 1500 })
    render()
  })

  it('matches snapshot', () => {
    const { asFragment } = render()
    expect(asFragment()).toMatchSnapshot()
  })
})
