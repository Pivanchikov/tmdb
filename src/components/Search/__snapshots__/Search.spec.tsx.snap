// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchDecstop matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-searchContainer-17"
  >
    <select
      data-testid="select"
      id="uc"
    >
      <option
        value="283566"
      >
        Evangelion: 3.0+1.0 Thrice Upon a Time
      </option>
      <option
        value="337404"
      >
        Cruella
      </option>
      <option
        value="379686"
      >
        Space Jam: A New Legacy
      </option>
      <option
        value="385128"
      >
        F9
      </option>
      <option
        value="436969"
      >
        The Suicide Squad
      </option>
    </select>
    <button
      class="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-18"
      data-testid="buttonSearch"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          data-testid="searchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
